src/tests/ftest/util/ssh_connection.py

```diff
@@ -0,0 +1,137 @@
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+
+from paramiko import client
+
+class Ssh(object):
+    """
+    It's mimic to the subprocess class, only difference is to use ssh protocol
+    and get command execution on remote machine where in subprocess command will
+    be initiated on local machine.
+    Right now check_output() and call() function implemented.In future more
+    methods can be added which will behave like subprocess methods.
+    """
+    def __init__(self, address, timeout=60, debug=False):
+        """
+        Initialize the remote machines and open SSH Connection.
+        Args:
+            address: remote machine IP address or hostname.
+            timeout: Timeout for waiting to finish the command
+            debug  : To print the command on console.
+        return:
+            None
+        """
+        self.address = address
+        self.timeout = timeout
+        self.debug = debug
+        self.client = client.SSHClient()
+        self.client.set_missing_host_key_policy(client.AutoAddPolicy())
+
+    def connect(self):
+        """
+        Connect the client via ssh connection
+        """
+        self.client.connect(self.address)
+
+        if self.debug:
+            print("SSH Connection open to {0}.".format(self.address))
+
+    def check_output(self, command):
+        """
+        Execute command on remote machine.
+        Args:
+            command: Command to run on remote machine.
+        return:
+            stdout: stdout of the return code
+        Raises:
+            general: if it's failed in ssh channel setup, exec command, close()
+            general: If command exit status is not 0.
+        """
+        try:
+            transport = self.client.get_transport()
+            transport.set_keepalive(20)
+            channel = transport.open_session()
+            channel.settimeout(self.timeout)
+            channel.exec_command(command)
+            stdout = channel.makefile('rb').read()
+            exit_status = channel.recv_exit_status()
+            channel.close()
+        except:
+            print("Exception in call for command {0}".format(command))
+            raise
```

Comment:
Ideally we should raise a known exception type such that consumers of this function can write expectant try/excepts

Reply:
Let's say for argument's sake that there are 5 different exceptions that could be raised in this try-except, and that from the standpoint of the `check_output()` function they essentially mean the same thing - we had a problem getting the exit status from the command.  If the function just passes the raised exception along, then each time a developer uses this function they need to write code to handle those 5 exceptions - and that assumes that the developer dove past this code into paramiko code to figure out the names of each of the exceptions.  I would argue it will save time and result in more robust code (no missed exceptions) if that specialized knowledge of the 5 exceptions is handled here and then the consumer of this function only has to handle the one exception - something went wrong trying to get the command status.

Reply:
Expectation was to pass the ssh exception as it is to the user level instead of adding any additional grouping of exception.

Reply:
But @phender makes a pretty good argument why this is not so good, primarily that every consumer needs to understand and handle the same bunch of paramiko exceptions, none of which they probably particularly care about and will handle all in the same way.

Reply:
ok will catch all exception as mention by @phender and pass it to the user level.